src/coreclr/src/jit/codegenarmarch.cpp

```diff
@@ -1780,7 +1780,39 @@ void CodeGen::genCodeForLclFld(GenTreeLclFld* tree)
     unsigned varNum = tree->GetLclNum();
     assert(varNum < compiler->lvaCount);
 
-    emit->emitIns_R_S(ins_Load(targetType), emitActualTypeSize(targetType), targetReg, varNum, offs);
+    emitAttr    attr = emitActualTypeSize(targetType);
+    instruction ins  = ins_Load(targetType);
+
+#ifdef TARGET_ARM
+    if (varTypeIsFloating(targetType) && ((offs % emitTypeSize(TYP_FLOAT)) != 0))
+    {
+        // Arm supports unaligned access only for integer types,
+        // load the floating data as 2 integer registers and convert them to float.
+        regNumber addr = tree->ExtractTempReg();
+        emit->emitIns_R_S(INS_lea, attr, addr, varNum, offs);
+
+        if (targetType == TYP_FLOAT)
+        {
+            regNumber floatAsInt = tree->GetSingleTempReg();
+            emit->emitIns_R_R(INS_ldr, EA_4BYTE, floatAsInt, addr);
+            emit->emitIns_R_R(INS_vmov_i2f, EA_4BYTE, targetReg, floatAsInt);
+        }
+        else
+        {
+            regNumber halfdoubleAsInt1 = tree->ExtractTempReg();
```

Comment:
Same here - could potentially be one instruction less
```suggestion
            regNumber halfdoubleAsInt1 = tree->ExtractTempReg();
            regNumber halfdoubleAsInt2 = tree->GetSingleTempReg();
            assert (halfdoubleAsInt1 < halfdoubleAsInt2);
            regMaskTP doubleAsIntMask = genRegMask(halfdoubleAsInt1) | genRegMask(halfdoubleAsInt2);
            emit->emitIns_R_I(INS_ldm, EA_4BYTE, addr, doubleAsIntMask);
            emit->emitIns_R_R_R(INS_vmov_d2i, EA_8BYTE, halfdoubleAsInt1, halfdoubleAsInt2, dataReg);
```
However, I am not sure how well instruction ldm is supported by emitter - I looked briefly and couldn't find an implementation for this in emitarm.cpp.